packages/devp2p/src/types.ts

```diff
@@ -1,8 +1,66 @@
 import type { DPT } from './dpt/index.js'
+import type { ETH } from './protocol/eth.js'
+import type { LES } from './protocol/les.js'
 import type { Protocol } from './protocol/protocol.js'
+import type { SNAP } from './protocol/snap.js'
+import type { Peer } from './rlpx/peer.js'
 import type { Common } from '@ethereumjs/common'
+import type { NestedUint8Array } from '@ethereumjs/rlp'
 import type { Socket } from 'net'
 
+export interface RLPxEvent {
+  'peer:added': [peer: Peer]
+  'peer:error': [peer: Peer, error: any]
+  'peer:removed': [peer: Peer, reason: any, disconnectWe: any] // disconnectWe indicates whether the disconnection was initiated by us or not
```

Comment:
I typed the disconnectWe as `boolean | null`. It was typed as that in rlpx/peer.ts and typecheck passes